### PR TITLE
97 Added database table and endpoint for average number of days for completion

### PR DESCRIPTION
**Pull request by jbackfieldVA**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): No*
- This is adding a public endpoint to return the average number of days to claim completion which will be stored in the database.
- VES

## Related issue(s)

- [[Claims] - Dev: Create Endpoint for Retrieving Average Numbers of Days#97](https://app.zenhub.com/workspaces/chatbot---benefit-microservices-65b953e667bb36095e6fd305/issues/gh/department-of-veterans-affairs/va-virtual-agent-benefits/97)
- Not a bug

## Testing done

- [x] *New code is covered by unit tests*
- Endpoint didn't exist, database table didn't exist
- Access the endpoint (with or without authentication) and receive a value.

## What areas of the site does it impact?
No other code was touched in this except the schema file to include the new table creation. This will be used for chatbot to display the average number of days to claim completion.

## Acceptance criteria

- [x]  I created the database table in a rails migration
- [x]  I updated the schema.rb file to include the new table
- [x]  Added unprotected endpoint to query the latest value
- [x]  We handle if there are no records in the database
- [x]  SPEC written and passes.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Requested Feedback

We are going to be creating a github action to insert records into the new table. This will only run ONCE per week so we are looking at a very small footprint of data in the database. There is a desire to keep the historical records in the database however, there is no discussion of data retention at the moment as the current schedule would mean that we would expect to run for 20 years before we even had 1000 records. If we need an index on the `created_at` in order to speed up the DB request, we can make that change.

### CODE REVIEW

**Comment by rmtolmach** [comment]
FYI: I updated your branch with `master` to fix the failing Danger check. It was fixed in #16317.

File: spec/controllers/v0/average_days_for_claim_completion_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    before do
+      AverageDaysForClaimCompletion.create(average_days: 100)
+      AverageDaysForClaimCompletion.create(average_days: 200)
+      AverageDaysForClaimCompletion.create(average_days: 300)
+    end
+
+    it 'returns the value' do
+      get :index
+      expect(response).to have_http_status(:ok)
+      expect(JSON.parse(response.body)['average_days']).to eq(300.0)
```
Comment (rmtolmach) [inline spec/controllers/v0/average_days_for_claim_completion_controller_spec.rb:36]: This test expects the average days with the highest number to be returned? If so, can you update the `it 'returns the value' do` to be more specific on what the expectation is?

**Review by bushnicoleVA — DISMISSED** [review]

File: spec/controllers/v0/average_days_for_claim_completion_controller_spec.rb (hunk 1 above)
Comment (jbackfieldVA) [reply spec/controllers/v0/average_days_for_claim_completion_controller_spec.rb:36]: It's supposed to return the most recent value in the database.

Comment (jbackfieldVA) [reply spec/controllers/v0/average_days_for_claim_completion_controller_spec.rb:36]: Should be better now.

**Review by bushnicoleVA — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**